tools/shared/src/main/scala/org/scalajs/core/tools/sourcemap/SourceMapWriter.scala

```diff
@@ -126,7 +151,8 @@ class SourceMapWriter(
     pendingPos = originalPos
     pendingName =
       if (startColumn != pendingColumnInGenerated) originalName
-      else pendingName orElse originalName
+      else if (pendingName != null) pendingName
+      else originalName
```

Comment:
Hum ... it seems this code is just plain wrong (I mean the existing code as much as the new code). The initial test
```scala
if (startColumn != pendingColumnInGenerated)
```
can never be true because we just did
```scala
pendingColumnInGenerated = startColumn
```
which means that once `pendingName` has been set once to non-null/`Some`, it will be forever unchanged. -_-'

I'm not even sure what's the purpose of these tests anymore :-s Anyway, let's not touch that in this PR, so that we don't hide change of behavior in the middle of a performance-oriented PR. I'll create an issue for this to investigate later.